capsules/extra/src/net/ipv6/ip_utils.rs

```diff
@@ -11,6 +11,8 @@ use crate::net::ieee802154::MacAddress;
 use crate::net::ipv6::IP6Header;
 use crate::net::udp::UDPHeader;
 
+pub const DEFAULT_DST_MAC_ADDR: MacAddress = MacAddress::Short(49138);
```

Comment:
Can you comment this?

Reply:
Sorry for the delay getting to these comments! This default mac address is the same as the default mac address in main.rs. The underlying (and deeper issue) is described in this comment in `components/udp_mux.rs`

        // In current design, all udp senders share same IP sender, and the IP
        // sender holds the destination mac address. This means all UDP senders
        // must send to the same mac address...this works fine under the
        // assumption of all packets being routed via a single gateway router,
        // but doesn't work if multiple senders want to send to different
        // addresses on a local network. This will be fixed once we have an
        // ipv6_nd cache mapping IP addresses to dst macs
        let ip_send =
            s.4.write(capsules_extra::net::ipv6::ipv6_send::IP6SendStruct::new(
                ip6_dg,
                ipsender_virtual_alarm,
                radio_buf,
                sixlowpan_tx,
                udp_mac,
                self.dst_mac_addr,
                self.src_mac_addr,
                ip_vis,
            ));

Joining a thread child requires first sending to 0xFFFF (multicast mac address) and then sending to the specific device address of the parent. We have two solutions to move forward:

1)  My _somewhat_ hacky solution to change the function calls to add the destination mac address (currently implemented but obviously confusing with default addrs etc)
2) Change the IPV6 capsule so that the destination mac address of the radio is changed based on the IPV6 address. This would revert some of the changes made to the udp capsule and remove the need for these defaults to be used)

@ppannuto & @bradjc  please let me know your thoughts

Reply:
Updating IPV6 is a much cleaner solution. I just made the changes and will push soon. This removes the need to change any UDP files and limits the changes to IPV6 to ~10 lines of code